src/client.js

```diff
@@ -150,9 +150,44 @@ module.exports = class Client extends EventEmitter {
     try {
       message = decrypt(object, this._credentials.keys);
     } catch (error) {
+      // NOTE(ibash) we seem to be getting errors while decrypting. I want to
+      // know if these errors are recoverable, whether it's just a single
+      // notification that will fail or if it's all notifications going forward.
+      // So we keep track of whether we've just seen a failed notification and
+      // whether we get a notification that goes through after that.
+
+      // Put the object in _persistentIds so we ignore a failed notification next time
+      this._persistentIds.push(object.persistentId);
+
+      // For detecting whether we can recover after failing to decrypt
+      this._failedToDecrypt = {object: object, keys: this._credentials.keys}
+
       error.metaData = {message: JSON.stringify(object), keys: JSON.stringify(this._credentials.keys)}
-      throw error
+      // throw in a setTimeout s.t. bugsnag can catch the error, but we don't
+      // want to break the connection
```

Comment:
Do we ever _want_ to break the connection?

Reply:
I think if we get an invalid message we'll want to. I think it was breaking before by throwing here.

Reply:
👍 cool as long as we break when we should